File: flutter/flutter_demo/lib/models/movie_model.dart
import 'package:json_annotation/json_annotation.dart';

part 'movie_model.g.dart';

@JsonSerializable(nullable: false)
class MovieModel {

  @JsonKey(name: "poster_path", required: false, nullable: true,)
  final String posterPath;

  @JsonKey(name: "adult", required: false, nullable: true,)
  final bool adult;

  @JsonKey(name: "overview", required: false, nullable: true,)
  final String overview;

  @JsonKey(name: "release_date", required: false, nullable: true,)
  final DateTime releaseDate;

  @JsonKey(name: "genre_ids", required: false, nullable: true,)
  final List<int> genreIds;

  @JsonKey(name: "original_title", required: false, nullable: true,)
  final String originalTitle;

  @JsonKey(name: "original_langauge", required: false, nullable: true,)
  final String originalLanguage;

  @JsonKey(name: "title", required: false, nullable: true,)
  final String title;

  @JsonKey(name: "backdrop_path", required: false, nullable: true,)
  final String backdropPath;

  @JsonKey(name: "popularity", required: false, nullable: true,)
  final double popularity;

  @JsonKey(name: "vote_count", required: false, nullable: true,)
  final int voteCount;

  @JsonKey(name: "video", required: false, nullable: true,)
  final bool video;

  @JsonKey(name: "vote_average", required: false, nullable: true,)
  final double voteAverage;

  MovieModel({this.posterPath, this.adult, this.overview, this.releaseDate, this.genreIds, this.originalTitle, this.originalLanguage, this.title, this.backdropPath, this.popularity, this.voteCount, this.video, this.voteAverage});

  factory MovieModel.fromJson(Map<String, dynamic> json) =>
      _$MovieModelFromJson(json);

  Map<String, dynamic> toJson() => _$MovieModelToJson(this);
}

File: android/app/src/main/java/com/example/fluttertalkandroid/models/MovieModel.java
package com.example.fluttertalkandroid.models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.Date;

public class MovieModel {
    @SerializedName("poster_path")
    private final String posterPath;

    @SerializedName("adult")
    private final Boolean adult;

    @SerializedName("overview")
    private final String overview;

    @SerializedName("release_date")
    private final Date releaseDate;

    @SerializedName("genre_ids")
    private final ArrayList<Integer> genreIds;

    @SerializedName("original_title")
    private final String originalTitle;

    @SerializedName("original_langauge")
    private final String originalLanguage;

    @SerializedName("title")
    private final String title;

    @SerializedName("backdrop_path")
    private final String backdropPath;

    @SerializedName("popularity")
    private final double popularity;

    @SerializedName("vote_count")
    private final int voteCount;

    @SerializedName("video")
    private final Boolean video;

    @SerializedName("vote_average")
    private final double voteAverage;

    public MovieModel(String posterPath, Boolean adult, String overview, Date releaseDate, ArrayList<Integer> genreIds, String originalTitle, String originalLanguage, String title, String backdropPath, double popularity, int voteCount, Boolean video, double voteAverage) {
        this.posterPath = posterPath;
        this.adult = adult;
        this.overview = overview;
        this.releaseDate = releaseDate;

        this.genreIds = genreIds;
        this.originalTitle = originalTitle;
        this.originalLanguage = originalLanguage;
        this.title = title;
        this.backdropPath = backdropPath;
        this.popularity = popularity;
        this.voteCount = voteCount;
        this.video = video;
        this.voteAverage = voteAverage;
    }

    public String getPosterPath() {
        return posterPath;
    }

    public Boolean getAdult() {
        return adult;
    }

    public String getOverview() {
        return overview;
    }

    public Date getReleaseDate() {
        return releaseDate;
    }

    public ArrayList<Integer> getGenreIds() {
        return genreIds;
    }

    public String getOriginalTitle() {
        return originalTitle;
    }

    public String getOriginalLanguage() {
        return originalLanguage;
    }

    public String getTitle() {
        return title;
    }

    public String getBackdropPath() {
        return backdropPath;
    }

    public double getPopularity() {
        return popularity;
    }

    public int getVoteCount() {
        return voteCount;
    }

    public Boolean getVideo() {
        return video;
    }

    public double getVoteAverage() {
        return voteAverage;
    }
}
